plugins/common/shim/output.go

```diff
@@ -0,0 +1,51 @@
+package shim
+
+import (
+	"bufio"
+	"fmt"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/parsers"
+)
+
+// AddOutput adds the input to the shim. Later calls to Run() will run this.
+func (s *Shim) AddOutput(output telegraf.Output) error {
+	if p, ok := output.(telegraf.Initializer); ok {
+		err := p.Init()
+		if err != nil {
+			return fmt.Errorf("failed to init input: %s", err)
+		}
+	}
+
+	s.Output = output
+	return nil
+}
+
+func (s *Shim) RunOutput() error {
+	// TODO: ? Need to support multiple parsers, but not clear how.
+	parser, err := parsers.NewInfluxParser()
```

Comment:
I would say no need, the shim abstracts the parsing aspect so you only need to deal with telegraf.Metric as a user.

Reply:
Would be nice for non-shim users who are more familiar with JSON, but that's a deeper topic for another time